mlflow/keras.py

```diff
@@ -385,33 +383,32 @@ class __MLflowKerasCallback(keras.callbacks.Callback):
         Records available logs after each epoch.
         Records model structural information as params when training begins
         """
+
         def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
-            try_mlflow_log(mlflow.log_param, 'num_layers', len(self.model.layers))
-            try_mlflow_log(mlflow.log_param, 'optimizer_name', type(self.model.optimizer).__name__)
+            try_mlflow_log(
+                mlflow.log_param, 'num_layers', len(
+                    self.model.layers))
+            try_mlflow_log(
+                mlflow.log_param, 'optimizer_name', type(
+                    self.model.optimizer).__name__)
             if hasattr(self.model.optimizer, 'lr'):
                 lr = self.model.optimizer.lr if \
-                    type(self.model.optimizer.lr) is float \
+                    isinstance(self.model.optimizer.lr, float) \
                     else keras.backend.eval(self.model.optimizer.lr)
                 try_mlflow_log(mlflow.log_param, 'learning_rate', lr)
             if hasattr(self.model.optimizer, 'epsilon'):
                 epsilon = self.model.optimizer.epsilon if \
-                    type(self.model.optimizer.epsilon) is float \
+                    isinstance(self.model.optimizer.epsilon, float) \
                     else keras.backend.eval(self.model.optimizer.epsilon)
                 try_mlflow_log(mlflow.log_param, 'epsilon', epsilon)
 
             sum_list = []
             self.model.summary(print_fn=sum_list.append)
             summary = '\n'.join(sum_list)
-            try_mlflow_log(mlflow.set_tag, 'model_summary', summary)
```

Comment:
We'd actually like to keep the `model_summary` as a tag for now. Instead of removing the tag completely to avoid the 5000 character limit, why don't we instead make a check for the length of the tag? If less than or equal to 5000 characters, we can log `model_summary` as a tag. If more than 5000 characters, we can instead log a tag that says `"Model summary too long to set as a tag. See the artifacts for full summary."` or something to that effect.

Additionally, would you be able to add a test that checks for this behavior? The test's location would be in `tests/keras_autolog/test_keras_autolog.py`.

Reply:
Thank you for the valuable comments! I fixed this behavior of `keras.py`, according to your suggestions [here](https://github.com/mlflow/mlflow/commit/e0fd234c02ecd8f3bfbf1d1960cf3159033f665f#diff-08cdd7de7cb6f904bb83e904b3e8fbf7R411-R424)

It now checks the length of the model summary. It sets it as a tag if the model's character length is less than 5000. If the model is larger, it sets as tag a message that it informs the user that it can't be set as a tag. The model is instead logged as an artifact in the artifacts' directory in the `model_summary.txt` file.


A test was also written based on the changes from @metazool in this [section](https://github.com/mlflow/mlflow/pull/2210/commits/d054b7c4e4b7cfcac4d62a32f0b28e2cab19b9c0#diff-eae3d2e34d3ebeb578b7c98a295e7727R136-R156)

A second large model had to be used to test this behavior and it's defined [here](https://github.com/mlflow/mlflow/pull/2210/commits/d054b7c4e4b7cfcac4d62a32f0b28e2cab19b9c0#diff-eae3d2e34d3ebeb578b7c98a295e7727R50)